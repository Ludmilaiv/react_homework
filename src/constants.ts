import { TAuthor } from './types';

export const AUTHOR: { BOT: TAuthor, USER: TAuthor } = {BOT: 'bot', USER: 'user'}; 