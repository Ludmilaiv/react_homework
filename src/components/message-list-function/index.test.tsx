import { MessageListFunction } from './';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { AUTHOR } from '../../constants';

describe('MessageListFunction', () => {
  it('render empty list', () => {
    render(<MessageListFunction messageList={[]} />);
  });

  it('render list', () => {
    const { asFragment } = render(<MessageListFunction messageList={[
      {id: '2', text: 'test text', authorName: 'test bot', author: AUTHOR.BOT}, 
      {id: '1', text: 'test bot text', authorName: 'test', author: AUTHOR.USER}]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render multiply components', () => {
    render(<>
      <MessageListFunction messageList={[]} />
      <MessageListFunction messageList={[]} />
    </>
    );

    expect(screen.queryAllByRole('list').length).toBe(2);
  });
});