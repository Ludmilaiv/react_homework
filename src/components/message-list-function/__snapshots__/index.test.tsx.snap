// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageListFunction render list 1`] = `
<DocumentFragment>
  <ul
    class="app__message-list"
    role="list"
  >
    <li>
      <b>
        test bot: 
      </b>
      <span>
        test text
      </span>
    </li>
    <li>
      <b>
        test: 
      </b>
      <span>
        test bot text
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
